internal/gps/source_manager.go

```diff
@@ -148,7 +148,7 @@ var _ SourceManager = &SourceMgr{}
 // gps's SourceManager is intended to be threadsafe (if it's not, please file a
 // bug!). It should be safe to reuse across concurrent solving runs, even on
 // unrelated projects.
-func NewSourceManager(cachedir string) (*SourceMgr, error) {
+func NewSourceManager(cachedir string, log func(v ...interface{})) (*SourceMgr, error) {
```

Comment:
injecting a logger into the SourceManager is something we need to do, but i've been delaying it for...well, not terribly good reasons. it feels kinda awkward to back it in this way, and especially to use a one-off type for doing it.

i'd be more comfortable, at least for now, with architecting out what we need internally, without actually changing the interface for it. sorta analogous to the top-level `context.Context` object we create in `NewSourceManager()`.

Reply:
👍 to shortening to `New()`.

is this config-method-to-create a common thing? don't think i've run across it before.

Reply:
I'm not sure. I hadn't really thought about it as a 'pattern', but it certainly seems to serve this case well and I'll keep it in mind in the future.

Reply:
i'm on the fence. it's basically making a factory out of the config type, which is...well, yeah, i'm on the fence.

can we go back to just `NewSourceManager(SourceManagerConfig{})` for now, and we can discuss this particular pattern in a standalone PR?

Reply:
Done.